### PR TITLE
Add fields to RateLimits struct

### PR DESCRIPTION
**Pull request by shokada**
Fixes: https://github.com/google/go-github/issues/2339

#### What this PR does / why we need it:
I added fields to `RateLimits` struct to get rate limits other than `core` and `search`

#### Special notes for your reviewer:
Since there is no documentation for `source_import`, `code_scanning_upload`, `actions_runner_registration`, and `scim`, I did not know which endpoints had the rate limit set, so I did not fix `category` function. 
https://github.com/google/go-github/blob/942d33d1d76e64a53d4d50ac0c95d54c907ba0e1/github/github.go#L1126-L1134

### CODE REVIEW

File: github/github.go
Comment (shokada) [inline github/github.go:1085]: It was 404, so I fixed it.

File: github/github.go
Comment (shokada) [inline github/github.go:1092]: It was 404, so I fixed it.

File: github/github.go
Comment (shokada) [inline github/github.go:1104]: There seems to be no document for `source_import`, `code_scanning_upload`, `actions_runner_registration`, `scim`.

File: github/github.go
Comment (gmlewis) [inline github/github.go:1096]:
```suggestion
	GraphQL *Rate `json:"graphql"`
```

File: github/github.go
Comment (gmlewis) [inline github/github.go:1104]:
```suggestion
	SCIM                      *Rate `json:"scim"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @shokada!
Let's tweak some of the acronym names, please.
Then we should be ready for a second LGTM+Approval before merging.

**Comment by gmlewis** [comment]
Also please remember to run `gofmt` on your edited files and `go generate ./...` , then push (not force-push) the changed files to the PR.

File: github/github.go
Comment (shokada) [reply github/github.go]: fixed https://github.com/google/go-github/pull/2340/commits/8b2ba544c2292ac7e432ca2df2116aa5ccc59084

File: github/github.go
Comment (shokada) [reply github/github.go]: fixed https://github.com/google/go-github/pull/2340/commits/8b2ba544c2292ac7e432ca2df2116aa5ccc59084

**Comment by shokada** [comment]
run `go generate ./...` https://github.com/google/go-github/pull/2340/commits/5887f39ce8055e1a8379a4e31e42069f2388f430

File: github/github.go
Comment (shokada) [inline github/github.go:1121]: Since `TestClient_rateLimits` does not pass, I have defined unused constants.

Comment (gmlewis) [reply github/github.go:1121]: Instead of doing this, we actually need to use these values in the `category` function (lines 1127-1134 below) and in the `RateLimits` method (lines 1154-1163 below).

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/github.go
Comment (shokada) [reply github/github.go]: > in the RateLimits method (lines 1154-1163 below).

fixed https://github.com/google/go-github/pull/2340/commits/a6c7bcd7f8602717e83b98d98444c9da8a7920d8

> in the category function (lines 1127-1134 below)

I could not figure out how to fix the `category` function for the following reasons...

> Since there is no documentation for source_import, code_scanning_upload, actions_runner_registration, and scim, I did not know which endpoints had the rate limit set, so I did not fix category function.
https://github.com/google/go-github/pull/2340#issue-1213489146

**Review by gmlewis — APPROVED** [review]
Thank you, @shokada !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
> LGTM

Thank you, @raynigon !
Merging.

**Comment by patrickmarabeas** [comment]
@gmlewis is it possible to get a fresh release pushed out? Thanks!

**Comment by gmlewis** [comment]
Yes, I can work on one tomorrow.

**Comment by gmlewis** [comment]
> @gmlewis is it possible to get a fresh release pushed out? Thanks!

@patrickmarabeas - `v45` has now been released: https://github.com/google/go-github/releases/tag/v45.0.0

**Pull request merged**
